includes/ucf-academic-calendar-feed.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Handles retrieving academic calendar data
+ **/
+if ( ! class_exists( 'UCF_Acad_Cal_Feed' ) ) {
+	class UCF_Acad_Cal_Feed {
+		/**
+		 * Retrieves the academic calendar feed
+		 * @author Jim Barnes
+		 * @since 1.0.0
+		 * @param array $args Arguments to override the configured default
+		 * @return array The academic calendar events in an array
+		 **/
+		public static function get_feed( $args ) {
+			// Get default values
+			$args = shortcode_atts(
+				UCF_Acad_Cal_Config::get_option_defaults(),
+				$args
+			);
+
+			$feed_url       = $args['calendar_feed'];
+			$use_cache      = $args['cache_items'];
+			$items          = false;
+
+			if ( $use_cache ) {
+				$transient_name = self::get_transient_name( $feed_url );
+				$items = get_transient( $transient_name );
+			}
+
+			if ( $items === false ) {
+				$response = wp_remote_get( $feed_url, array( 'timeout' => 15 ) );
+
+				if ( is_array( $response ) ) {
+					$retobj = json_decode( wp_remote_retrieve_body( $response ) );
+
+					if ( is_array( $retobj->terms ) ) {
+						$items = $retobj->terms[0]->events;
+					}
+				}
+
+				if ( $use_cache ) {
+					$expiration = $args['cache_expiration'] * HOUR_IN_SECONDS;
+
+					set_transient( $transient_name, $items, $expiration );
+				}
+			}
+
+			if ( $items ) {
+				$items = array_slice( $items, 0, $args['default_count'] );
```

Comment:
Not necessary for this PR, but having an 'offset' arg that we could use here could be helpful--we may need two separate set of feed results to achieve the layout we need on the main site Students page, and being able to set an offset would let us display the two separately without repeating the first calendar item.

Reply:
Will take care of this for the "Expanded Options" milestone, issue #2 .